agent/exec/dockerapi/container.go

```diff
@@ -432,7 +432,8 @@ func (c *containerConfig) volumeCreateRequest(mount *api.Mount) *volume.VolumeCr
 		labels = mount.VolumeOptions.Labels
 	}
 
-	return &volume.VolumeCreateBody{
+	// FIXME: do we need the ClusterVolumeSpec here?
```

Comment:
Ugh, it looks like we should pass ClusterVolumeSpec for API >=1.42

https://github.com/moby/moby/blob/aac5e5906d31d339fd9ed67fe9198d2444efa7d7/api/server/router/volume/volume_routes.go#L106-L124

20.10.3 seems to be API 1.41. So it seems like we need that?

Reply:
I'm not familiar with this code though 🙈

Reply:
Yeah, it’s a new feature, and at this time “not yet supported” on the executor implementation in SwarmKit itself. 

The implementation in docker lives in https://github.com/moby/moby/blob/v23.0.0-beta.1/daemon/cluster/executor/container/controller.go

So yes, it should be added at some point (perhaps I should create a tracking ticket for it) for swarmkit running standalone as well, but first priority for me was to make sure it is at least compatible with the upcoming (v23) release.

Reply:
Opened https://github.com/moby/swarmkit/issues/3104

Reply:
Ah, thanks for the explanation!